modules/core-updates/9.0.md

```diff
@@ -78,7 +78,91 @@ Some dependencies are still present but were upgraded which comes with their own
+     * 
+     * @param MyCustomService $myCustomService
+     */
+    public function __construct(
+      private readonly MyCustomService $myCustomService,
+    ) {
+    }
+    
+    /**
+     * You can inject a service only on one method, useful if the service is used only in one action, it is more optimized than injecting it all the time and never use it.
+     * 
+     * @param MySpecificService $specificService
+     * @return JsonResponse
+     */
+    public function indexAction(MySpecificService $specificService): JsonResponse
+    {
+        $generalData = $this->myCustomerService->getData();
+        $specificData = $specifiService->getSpecificData();
+
+        // You can use the get method to fetch a service that was added in the getSubscribedServices method
+        $toolingService = $this->container->get(ToolingService::class);
+        
+        return new JsonResponse($toolingService->format($generalData + $specificData));
+    }
+    
+    /**
+     * You can define a list of registered services, they will be accessible via the $this->>container->get method, this
+     * way of injecting service is interesting if implemented on a base controller class for generic services that
+     * may be used by many controllers (like the router, the translator, ...) so you don't need to inject them manually
+     * in each controller class.
+     * 
+     * @return array
+     */
+    public static function getSubscribedServices(): array
+    {
+        return return parent::getSubscribedServices() + [
+            ToolingService::class => ToolingService::class,
+        ];
+    }
+}
+```
+
+**Define your controller as a service**
+
+You can already anticipate and define your controllers as services, it is also compatible with previous versions of Symfony and PrestaShop:
+
+```yaml
+services:
+  MyNamespace\MyController:
+    # Using autowire will inject services in the constructors automatically
+    autowire: true
+    # Using autoconfigure with controllers that extend AbstractController will handle method parameter injection and subscribed services
+    autoconfigure: true
+    # If your controller class doesn't extend AbstractController you should add this tag manually si you benefit from method parameters injection
+    tags: ['controller.service_arguments']
+```
+
+{{% notice note %}}
+**PrestaShopAdminController::getSubscribedServices**
+
+The new base controller class uses the `getSubscribedServices` method to give easy access to some commonly used PrestaShop services, like the `ConfigurationInterface`, `HookDispatcherInterface`, `TranslatorInterface`, ...
+You can see the <a href="https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Controller/Admin/PrestaShopAdminController.php#L59" target="_blank">full list of subscribed services</a> in the class, this list will probably grow piece by piece but only for common generic services.
```

Comment:
```suggestion
You can see the <a href="https://github.com/PrestaShop/PrestaShop/blob/9.0.0-alpha.1/src/PrestaShopBundle/Controller/Admin/PrestaShopAdminController.php#L59" target="_blank">full list of subscribed services</a> in the class, this list will probably grow piece by piece but only for common generic services.
```

we'll change to v9 final once ready

Reply:
Yeah that's what I though as well, that's why I used the branch for now But okay for that tag